source/camel-cased-properties-deep.d.ts

```diff
@@ -44,11 +45,43 @@ const result: CamelCasedPropertiesDeep<UserWithFriends> = {
 @category Template literal
 @category Object
 */
-export type CamelCasedPropertiesDeep<Value, Options extends CamelCaseOptions = {preserveConsecutiveUppercase: true}> = Value extends Function
+export type CamelCasedPropertiesDeep<
+	Value,
+	Options extends CamelCaseOptions = {preserveConsecutiveUppercase: true},
+> = Value extends Function
 	? Value
-	: Value extends Array<infer U>
-		? Array<CamelCasedPropertiesDeep<U, Options>>
+	: Value extends UnknownArray
+		? CamelCasedPropertiesArrayDeep<Value>
 		: Value extends Set<infer U>
-			? Set<CamelCasedPropertiesDeep<U, Options>> : {
-				[K in keyof Value as CamelCase<K, Options>]: CamelCasedPropertiesDeep<Value[K], Options>;
+			? Set<CamelCasedPropertiesDeep<U, Options>>
+			: {
+				[K in keyof Value as CamelCase<K, Options>]: CamelCasedPropertiesDeep<
+				Value[K],
+				Options
+				>;
 			};
+
+// This is a copy of DelimiterCasedPropertiesArrayDeep (see: delimiter-cased-properties-deep.d.ts).
+// These types should be kept in sync.
```

Comment:
You need to add this text to `DelimiterCasedPropertiesArrayDeep` too, otherwise it could get out of sync with this one.